builder/azure/arm/config.go

```diff
@@ -1256,6 +1263,13 @@ func assertRequiredParametersSet(c *Config, errs *packersdk.MultiError) {
 		if c.SharedGalleryDestination.SigDestinationSubscription == "" {
 			c.SharedGalleryDestination.SigDestinationSubscription = c.ClientConfig.SubscriptionID
 		}
+
+		if c.SharedGalleryDestination.SigDestinationUseShallowReplicationMode {
+			// When the user doesn't set the replica count it uses the int null value (0) so we have to check for that
+			if c.SharedGalleryImageVersionReplicaCount != 1 && c.SharedGalleryImageVersionReplicaCount != 0 {
+				errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("When using shallow replication the replica count can only be 1, leaving this value unset will default to 1"))
+			}
+		}
```

Comment:
I think we can replaced the comment by setting the count to 1 if not value is specified. 
```suggestion

		if c.SharedGalleryDestination.SigDestinationUseShallowReplicationMode && c.SharedGalleryImageVersionReplicaCount  == 0 {
		c.SharedGalleryImageVersionReplicaCount = 1
		}
			
		if c.SharedGalleryImageVersionReplicaCount > 1 {
				errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("When using shallow replication the replica count can only be 1, leaving this value unset will default to 1"))
			}
		}
```